R/build_email_request.R

```diff
@@ -40,28 +40,39 @@ build_email_request.blastula_message <- function(body, content_type,
 
 build_email_request.envelope <- function(body, token=NULL, user_id=NULL, ...)
 {
+    require_emayili_0.6()
     parts <- body$parts
 
-    inline <- which(sapply(parts, function(p) p$header$content_disposition == "inline"))
+    # parts is either a single body object (itself a named list), or a list of body objects
+    if(!is.null(names(parts)))
+        parts <- list(parts)
+
+    inline <- which(sapply(parts, function(p) p$disposition == "inline"))
     if(length(inline) > 1)
         warning("Multiple inline sections found, only the first will be used", call.=FALSE)
     req <- if(!is_empty(inline))
     {
         inline <- parts[[inline[1]]]
         list(
             body=list(
-                contentType=if(inline$header$content_type == "text/html") "html" else "text",
-                content=inline$body
+                contentType=if(inherits(inline, "text_html")) "html" else "text",
+                content=inline$content
             )
         )
     }
     else list(body=list(contentType="text", content=""))
 
     if(!is_empty(body$header$Subject))
-        req$subject <- body$header$Subject
+        req$subject <- body$header$Subject$values
+
+    recipients <- build_email_recipients(
+        unclass(body$headers$To$values)$email,
+        unclass(body$headers$Cc$values)$email,
+        unclass(body$headers$Bcc$values)$email,
+        unclass(body$headers$`Reply-To`$values)$email
```

Comment:
@datawookie can you verify that this is the correct way to handle addresses? The `build_email_recipients` function expects its inputs to be character vectors, and I just discovered that emayili uses a custom address class.

Reply:
Yes. That will work. Or you can use the accessor functions `to()`, `cc()` etc.

```r
library(emayili)

msg <- envelope(
  to = c("anne@example.com", "bob@example.com")
)

unclass(msg$headers$To$values)$email
#> [1] "anne@example.com" "bob@example.com"
as.character(to(msg))
#> [1] "anne@example.com" "bob@example.com"
```